#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_use_tempfile_by_default.dpatch by Stefan Fritsch <sf@sfritsch.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Using a tempfile is a lot faster than direct writes on ReiserFS
## DP: and has no disadvantages besides diskspace usage.
## DP: => We make it the default and let -T restore the old behaviour.

@DPATCH@
diff -urNad mp3gain-1.5.1~/mp3gain.c mp3gain-1.5.1/mp3gain.c
--- mp3gain-1.5.1~/mp3gain.c	2009-07-24 15:28:57.950388245 +0200
+++ mp3gain-1.5.1/mp3gain.c	2009-07-24 15:30:31.702394680 +0200
@@ -45,6 +45,9 @@
  *  So have fun dissecting.
  */
 
+/* stf: make 	-t the default
+		-T restores old behaviour */
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
@@ -129,7 +132,7 @@
 
 int writeself = 0;
 int QuietMode = 0;
-int UsingTemp = 0;
+int UsingTemp = 1;
 int NowWriting = 0;
 double lastfreq = -1.0;
 
@@ -1045,7 +1048,7 @@
             deleteFile(outfilename);
 			passError( MP3GAIN_UNSPECIFED_ERROR, 3,
                 "Not enough temp space on disk to modify ", filename, 
-                "\nEither free some space, or do not use \"temp file\" option\n");
+                "\nEither free some space, or switch off \"temp file\" option with -T\n");
             return M3G_ERR_NOT_ENOUGH_TMP_SPACE;
         }
         else {
@@ -1326,10 +1329,11 @@
 		fprintf(stderr,"\t%cd <n> - modify suggested dB gain by floating-point n\n",SWITCH_CHAR);
 		fprintf(stderr,"\t%cc - ignore clipping warning when applying gain\n",SWITCH_CHAR);
 		fprintf(stderr,"\t%co - output is a database-friendly tab-delimited list\n",SWITCH_CHAR);
-		fprintf(stderr,"\t%ct - writes modified data to temp file, then deletes original\n",SWITCH_CHAR);
-		fprintf(stderr,"\t     instead of modifying bytes in original file\n");
+		fprintf(stderr,"\t%ct - mp3gain writes modified mp3 to temp file, then deletes original \n",SWITCH_CHAR);
+		fprintf(stderr,"\t     instead of modifying bytes in original file (default)\n");
+		fprintf(stderr,"\t%cT - mp3gain directly modifies mp3 file (opposite of %ct)\n",SWITCH_CHAR,SWITCH_CHAR);
 #ifdef AACGAIN
-		fprintf(stderr,"\t     A temp file is always used for AAC files.\n");
+		fprintf(stderr,"\t     Ignored for AAC files.\n");
 #endif
 		fprintf(stderr,"\t%cq - Quiet mode: no status messages\n",SWITCH_CHAR);
 		fprintf(stderr,"\t%cp - Preserve original file timestamp\n",SWITCH_CHAR);
@@ -1673,10 +1677,13 @@
                     break;
 
 				case 't':
-				case 'T':
 					UsingTemp = !0;
 					break;
 
+				case 'T':
+					UsingTemp = 0;
+					break;
+
 				case 'u':
 				case 'U':
 					undoChanges = !0;
